cg/apps/lims/api.py

```diff
@@ -85,74 +89,96 @@ def _export_artifact(self, lims_artifact):
+
+        step_names_udfs = MASTER_STEPS_UDFS['prepared_step']
+        prepared_dates = []
+
+        for process_type in step_names_udfs:
+            artifacts = self.get_artifacts(process_type=process_type, samplelimsid=lims_id)
+
+            for artifact in artifacts:
+                prepared_dates.append(parse_date(artifact.parent_process.date_run))
+
+        if prepared_dates:
+            sorted_dates = sorted(prepared_dates, reverse=True)
+            prepared_date = sorted_dates[0]
+
+        return prepared_date if prepared_dates else None
 
     def get_delivery_date(self, lims_id: str) -> dt.date:
         """Get delivery date for a sample."""
-        artifacts = self.get_artifacts(samplelimsid=lims_id, process_type='Delivery v1',
-                                       type='Analyte')
-        if len(artifacts) == 1:
-            return artifacts[0].parent_process.udf.get('Date delivered')
-        elif len(artifacts) == 0:
-            return None
-        else:
-            log.warning(f"multiple delivery artifacts found for: {lims_id}")
-            return min(artifact.parent_process.udf['Date delivered'] for artifact in artifacts)
+
+        step_names_udfs = MASTER_STEPS_UDFS['delivery_step']
+
+        delivered_dates = self._get_all_step_dates(step_names_udfs, lims_id,
+                                                   artifact_type='Analyte')
+
+        if len(delivered_dates) > 1:
+            log.warning("multiple delivery artifacts found for: %s", lims_id)
+
+        delivered_date = self._most_recent_date(delivered_dates)
+
+        return delivered_date
 
     def get_sequenced_date(self, lims_id: str) -> dt.date:
         """Get the date when a sample was sequenced."""
-        process_types = ['CG002 - Illumina Sequencing (Illumina SBS)',
-                         'CG002 - Illumina Sequencing (HiSeq X)']
-
-        for process_type in process_types:
-            lims_artifacts = self.get_artifacts(process_type=process_type, samplelimsid=lims_id)
-
-            if len(lims_artifacts) == 0:
-                continue  # continue in case the artifacts are in the other process_type
-            elif len(lims_artifacts) == 1:
-                return lims_artifacts[0].parent_process.udf.get('Finish Date')  # return date immediately if only one artifact is found
-            else:
-                log.warning(f"multiple sequence artifacts found for: {lims_id}")
-                sequence_dates = [artifact.parent_process.udf.get('Finish Date') for artifact in lims_artifacts
-                                  if artifact.parent_process.udf.get('Finish Date') is not None]
-                if len(sequence_dates) == 0:
-                    break  # nothing sequenced, break out and return None
-                else:
-                    return min(sequence_dates)
-
-        return None  # if nothing is found just return None
+        novaseq_process = 'AUTOMATED - NovaSeq Run'
```

Comment:
Would something like this also not look great in a constants file?

Reply:
Yeah, how about https://github.com/Clinical-Genomics/cg/pull/376/commits/0c47a84f5270eb798e667b41e6167d5da5545799?

Reply:
Perfect!